pkg/k8s/network_policy.go

```diff
@@ -146,17 +146,21 @@ func ParseNetworkPolicy(np *networkingv1.NetworkPolicy) (api.Rules, error) {
 	}
 
 	// Convert the k8s default-deny model to the Cilium default-deny model
-	// spec:
-	//   policyTypes:
-	//   - ingress
-	if len(ingresses) == 0 && hasV1PolicyType(np.Spec.PolicyTypes, networkingv1.PolicyTypeIngress) {
+	//spec:
+	//  podSelector: {}
+	//  policyTypes:
+	//	  - Ingress
+	// Since k8s 1.7 doesn't contain any PolicyTypes, we default deny
+	// if podSelector is empty and the policyTypes is not egress
+	if len(ingresses) == 0 && !hasV1PolicyType(np.Spec.PolicyTypes, networkingv1.PolicyTypeEgress) {
```

Comment:
This is not correct, it does not account for the "ingress, egress" case.

I think it needs to be this:
``` go
if len(ingresses) == 0 &&
   (hasV1PolicyType(np.Spec.PolicyTypes, networkingv1.PolicyTypeIngress) || !hasV1PolicyType(np.Spec.PolicyTypes, networkingv1.PolicyTypeEgress)) {
```

We need to enable ingress default deny in the following cases:
 - `policyTypes` contains `ingress`
 - `policyTypes` is emitted entirely.

Reply:
Right, fixed.